core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -191,6 +194,16 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
 
       // Bootstrap to fetch the driver's Spark properties.
       val executorConf = new SparkConf
+
+      if (System.getenv(SecurityManager.ENV_AUTH_SECRET) != null) {
+        executorConf.set("spark.authenticate", "true")
+        val secret = System.getenv(SecurityManager.ENV_AUTH_SECRET)
+        if (Files.exists(Paths.get(secret))) {
```

Comment:
I know this is a cherry-pick, but just so that I understand a bit more clearly. What is the difference between this and the code for: [`getFileBasedSecret`](https://github.com/mesosphere/spark/pull/38/files#diff-afd88f677ec5ff8b5e96a5cbbe00cd98R528)?

Reply:
It's a bit different, but the file contents reading part is the same, right. This code definitely needs some refactoring but I think it's a good idea to avoid mixing of cherry-picking with code changes in case any of these commits go to the upstream. Probably, it's even better to create a clean patch to the upstream and then backport it to this repo.